server/blob/blob-objectstorage/src/test/java/org/apache/james/blob/objectstorage/ObjectStorageBlobsDAOAWSTest.java

```diff
@@ -53,17 +63,22 @@ void setUp(DockerAwsS3Container dockerAwsS3) {
         ObjectStorageBlobsDAOBuilder.ReadyToBuild builder = ObjectStorageBlobsDAO
             .builder(configuration)
             .defaultBucketName(defaultBucketName)
-            .blobIdFactory(new TestBlobId.Factory())
+            .blobIdFactory(BLOB_ID_FACTORY)
             .putBlob(awsS3ObjectStorage.putBlob(configuration));
 
         blobStore = builder.getSupplier().get();
         objectStorageBlobsDAO = builder.build();
         objectStorageBlobsDAO.createBucket(defaultBucketName).block();
+        testee = new MetricableBlobStore(metricsTestExtension.getMetricFactory(), objectStorageBlobsDAO);
     }
 
     @AfterEach
     void tearDown() {
-        blobStore.deleteContainer(defaultBucketName.asString());
+        blobStore.list()
+            .stream()
+            .filter(storageMetadata -> storageMetadata.getType().equals(StorageType.CONTAINER))
+            .map(StorageMetadata::getName)
+            .forEach(containerName -> blobStore.deleteContainer(containerName));
```

Comment:
1. do a distinvct
2. I bet we do only use 2-3 containers in total. Maybe hardcoding their delete here will be better?